src/shared/utils/utils.test.ts

```diff
@@ -263,4 +277,68 @@ describe("Util Functions", () => {
+    });
+    test("should be not be valid when document is an empty object", () => {
+      expect(isWrappedV2Document({})).toBe(false);
+    });
+    test("should be not be valid when document is null", () => {
+      expect(isWrappedV2Document(null)).toBe(false);
+    });
+    test("should be not be valid when document.data is missing", () => {
+      expect(isWrappedV2Document(omit(cloneDeep(wrappedV2Document), "data"))).toBe(false);
+    });
+    test("should be not be valid when document.data is null", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "data", null))).toBe(false);
+    });
+    test("should be not be valid when document.data.issuers is missing", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "data.issuers", null))).toBe(false);
+    });
+    test("should be not be valid when document.signature is an empty object", () => {
+      expect(isWrappedV2Document(omit(cloneDeep(wrappedV2Document), "signature"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.type is missing", () => {
+      expect(isWrappedV2Document(omit(cloneDeep(wrappedV2Document), "signature.type"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.type is invalid", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "signature.type", "oops"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.targetHash is missing", () => {
+      expect(isWrappedV2Document(omit(cloneDeep(wrappedV2Document), "signature.targetHash"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.targetHash is invalid", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "signature.targetHash", "oops"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.merkleRoot is missing", () => {
+      expect(isWrappedV2Document(omit(cloneDeep(wrappedV2Document), "signature.merkleRoot"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.merkleRoot is invalid", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "signature.merkleRoot", "oops"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.proof is missing", () => {
+      expect(isWrappedV2Document(omit(cloneDeep(wrappedV2Document), "signature.proof"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.proof is a string", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "signature.proof", "oops"))).toBe(false);
+    });
+    test("should be not be valid when document.signature.proof is an array of number", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "signature.proof", [2]))).toBe(false);
+    });
+    test("should be not be valid when document.signature.proof is array of hash", () => {
```

Comment:
"should be valid" instead of "should not"

Reply:
also should not be valid instead of should be not

Reply:
i didnt even catch that 😓

Reply:
done